cmd/bugzilla-config-manager/main.go

```diff
@@ -0,0 +1,145 @@
+	var errs []error
+	var developmentVersionFound bool
+
+	for ocpProduct, productConfig := range oracle {
+		if ocpProduct != "ocp" {
+			continue
+		}
+
+		for productVersion, config := range productConfig {
+			if len(config.Previous) < 1 {
+				continue
+			}
+
+			// TODO: We assume the first event is the last - Is that reasonable?
+			switch config.Previous[0].Event {
+			case LifecycleEventGenerallyAvailable:
+				parsedVersion, err := parseMajorMinor(productVersion)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse %s as majorMinor version: %w", productVersion, err))
+					continue
+				}
+				gaVersions = append(gaVersions, *parsedVersion)
+			case LifecycleEventOpen:
+				if developmentVersionFound {
+					errs = append(errs, fmt.Errorf("found multiple development versions: %s and %s", developmentVersion, productVersion))
+				}
+				parsedVersion, err := parseMajorMinor(productVersion)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse %s as majorMinor version: %w", productVersion, err))
+					continue
+				}
+				developmentVersion = *parsedVersion
+				developmentVersionFound = true
+			}
+		}
+	}
+
+	if !developmentVersionFound {
+		errs = append(errs, errors.New("didn't find a development version in config"))
+	}
+
+	return developmentVersion, gaVersions, utilerrors.NewAggregate(errs)
+}
+
+func parseMajorMinor(version string) (*MajorMinor, error) {
+	dotSplit := strings.Split(version, ".")
+	if len(dotSplit) != 2 {
+		return nil, fmt.Errorf("version %s split by dot doesn't have two elements, can't be in major.minor format", version)
+	}
+	parsedMajor, err := strconv.ParseInt(dotSplit[0], 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s as 32 bit base 10 int: %w", dotSplit[0], err)
+	}
+	parsedMinor, err := strconv.ParseInt(dotSplit[1], 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s as 32 bit base 10 int: %w", dotSplit[1], err)
+	}
+
+	return &MajorMinor{Major: int(parsedMajor), Minor: int(parsedMinor)}, nil
+}
+
+func reconcileConfig(cfg *plugins.Bugzilla, developmentVersion MajorMinor, gaVersions []MajorMinor) *plugins.Bugzilla {
+	if cfg.Default == nil {
+		cfg.Default = map[string]plugins.BugzillaBranchOptions{}
+	}
+
+	// main/master: Set developmentVersion and leave the rest as is
+	for _, developmentBranchName := range []string{"main", "master"} {
+		config := cfg.Default[developmentBranchName]
+		config.TargetRelease = utilpointer.String(developmentVersion.String() + "0")
+
+		cfg.Default[developmentBranchName] = config
+	}
+
+	// developmentVersion/developmentVersion+1: Set config for {release/openshift}-$VERSION:
+	// * dependentBugStatus: ["MODIFIED", "ON_QA", "VERIFIED"]
+	// * DependentBugTargetReleases: "$VERSION+1.0"
+	// * ValidateByDefault: true
+	for _, developmentVersion := range []MajorMinor{developmentVersion, developmentVersion.WithIncrementedMinor(1)} {
```

Comment:
I'm not really sure we should assume anything about the X.Y+1 one: if that's a version, it should be in a lifecycle with an "open" even some time in the future, and if we need special logic for that phase we should just have it instead of piggy backing on the development version.

Reply:
No strong feelings on that one way or another, it is just what I found in the existing config

Reply:
Note though that there can only be one version where `open` is the latest event, as that is the basis for configuring master/main so what is currently in our BZ config can't be modeled without adding X-1 logic to the tool.

Reply:
Yeah, just a single branch with `open` as latest past event, but it should be possible to have ones with `open` as a *future* event?

Reply:
So from the description in the card, future events are irrelevant for this configmanager, did I get that wrong?